tasks/image-segmentation-yolov8/src/validate.py

```diff
@@ -0,0 +1,238 @@
+from concurrent.futures import ProcessPoolExecutor, Future, as_completed
+from pathlib import Path
+
+import logging
+import gc
+import csv
+import multiprocessing as mp
+
+from PIL import Image, ImageDraw
+from coretex import TaskRun, ImageDataset, ImageDatasetClasses, folder_manager, TaskRunStatus
+from ultralytics import YOLO
+from matplotlib import pyplot as plt
+from ultralytics.engine.results import Results
+
+import numpy as np
+import torch
+
+
+def createDatasetResults(taskRun: TaskRun[ImageDataset], sampleResults: list[dict[str, str]]) -> None:
+    datasetResult: list[dict[str, str]] = []
+
+    for label in taskRun.dataset.classes.labels:
+        classAcc = [float(result[label]) for result in sampleResults]
+        datasetResult.append({
```

Comment:
I would not add this data to a list first, but rather call `writer.writerow` directly for this data